pkg/auth/authmap_cache.go

```diff
@@ -12,20 +12,24 @@ import (
 	"golang.org/x/exp/maps"
 
 	"github.com/cilium/cilium/pkg/lock"
+	"github.com/cilium/cilium/pkg/maps/authmap"
+	"github.com/cilium/cilium/pkg/metrics"
 )
 
 type authMapCache struct {
 	logger            logrus.FieldLogger
 	authmap           authMap
 	cacheEntries      map[authKey]authInfo
 	cacheEntriesMutex lock.RWMutex
+	pressureGauge     *metrics.GaugeWithThreshold
 }
 
-func newAuthMapCache(logger logrus.FieldLogger, authmap authMap) *authMapCache {
+func newAuthMapCache(logger logrus.FieldLogger, authMap authMap) *authMapCache {
 	return &authMapCache{
-		logger:       logger,
-		authmap:      authmap,
-		cacheEntries: map[authKey]authInfo{},
+		logger:        logger,
+		authmap:       authMap,
+		cacheEntries:  map[authKey]authInfo{},
+		pressureGauge: metrics.NewBPFMapPressureGauge(authmap.MapName, 0),
```

Comment:
is it somehow possible to use the new metrics approach from hive introduced by @dylandreimerink  ? -> https://docs.cilium.io/en/latest/contributing/development/hive/#metric

Reply:
It will be awesome to use the new pattern, just a note that this is the existing metric, which is still not migrated to the new hive and cell structure (i.e. adding the same metric will cause a double registration error in prometheus colllector).